src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs

```diff
@@ -227,14 +237,53 @@ public override void ConfigureServices(IServiceCollection collection)
             collection.TryAddSingleton<INupkgInfoFactory, NupkgInfoFactory>();
             collection.TryAddSingleton<IPackageArchiveReaderFactory, PackageArchiveReaderFactory>();
             collection.TryAddSingleton<IFileSystem, FileSystem>();
+            
+            // Register ProductionChannelValidator with the specified validation mode
+            collection.TryAddSingleton<ITargetChannelValidator>(provider =>
+            {
+                var buildInfoService = provider.GetRequiredService<IProductionChannelValidatorBuildInfoService>();
+                var branchClassificationService = provider.GetRequiredService<IBranchClassificationService>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var logger = loggerFactory.CreateLogger<ProductionChannelValidator>();
+                
+                // Convert EnforceProduction boolean to ValidationMode enum
+                var validationMode = EnforceProduction ? ValidationMode.Enforce : ValidationMode.Audit;
+                
+                return new ProductionChannelValidator(buildInfoService, branchClassificationService, logger, validationMode);
+            });
+            
+            // Add logging services
+            collection.AddLogging(builder => builder.SetMinimumLevel(LogLevel.Debug));
+            
+            // Register AzureDevOpsService with proper authentication
+            collection.TryAddSingleton<IProductionChannelValidatorBuildInfoService>(provider =>
+            {
+                var httpClient = provider.GetRequiredService<HttpClient>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var logger = loggerFactory.CreateLogger<AzureDevOpsService>();
+                return new AzureDevOpsService(httpClient, logger, AzdoApiToken);
+            });
+            
+            // Register BranchClassificationService with proper authentication
+            collection.TryAddSingleton<IBranchClassificationService>(provider =>
+            {
+                var httpClient = provider.GetRequiredService<HttpClient>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var logger = loggerFactory.CreateLogger<BranchClassificationService>();
+                return new BranchClassificationService(httpClient, logger, AzdoApiToken);
+            });
```

Comment:
I think it's a bit more resilient to future changes to do

```suggestion
            collection.TryAddSingleton<IProductionChannelValidatorBuildInfoService>(provider =>
                ActivatorUtilities.CreateInstance<AzureDevOpsService>(provider, AzdoApiToken));

            collection.TryAddSingleton<IBranchClassificationService>(provider =>
                ActivatorUtilities.CreateInstance<BranchClassificationService>(provider, AzdoApiToken));
```